CONTRIBUTING.md

```diff
@@ -243,18 +243,82 @@ If in doubt, you can always ask for guidance in the Pull Request or on
 [IRC in the #node-dev channel](https://webchat.freenode.net?channels=node-dev&uio=d4).
 
 Feel free to post a comment in the Pull Request to ping reviewers if you are
-awaiting an answer on something.
+awaiting an answer on something. If you encounter words or acronyms that
+seem unfamiliar, check out the [glossary](https://sites.google.com/a/chromium.org/dev/glossary).
 
+Note that your commits usually get squashed when they are landed (see
+notes about [commit squashing](#commit-squashing)).
 
 ### Step 8: Landing
 
-Once your Pull Request has been reviewed and approved by at least one Node.js
-Collaborators (often by saying LGTM, or Looks Good To Me), and as long as
-there is consensus (no objections from a Collaborator), a
-Collaborator can merge the Pull Request . GitHub often shows the Pull Request as
- `Closed` at this point, but don't worry. If you look at the branch you raised
- your Pull Request against (probably `master`), you should see a commit with
- your name on it. Congratulations and thanks for your contribution!
+In order to get landed, a Pull Request needs to be reviewed and
+[approved](#getting-approvals-for-your-pull-request) by
+at least one Node.js Collaborator. If this Pull Request touches
+more than the documentation, then it also needs an all-green
```

Comment:
Maybe `s/all-green/passing`. We often don't get an all-green CI run due to machine issues or flaky tests, but it's still counts as long as we can be confident that the failures were unrelated. I'm not sure it's worth explaining all that in this guide, `passing` should be fine.